GridList/GridList.jsx

```diff
@@ -0,0 +1,151 @@
+import { h } from "preact";
+import MaterialComponent from "../MaterialComponent";
+import Icon from "../Icon";
+
+const notEmptyString = val => val !== "";
+
+/**
+ * @prop tile-gutter {1|4} - Number of px for the padding/spacing between items.
+ * @prop header-caption {boolean} - position <GridList.Tile.Secondary> at top
+ * @prop twoline-caption {boolean} - add spacing to <GridList.Tile.Secondary> for <GridList.Tile.SupportText>
+ * @prop with-icon-align {"start"|"end"} - position <GridList.Tile.Icon> at beginning or end of <GridList.Tile.Secondary>
+ * @prop tile-aspect-ratio {"1x1"|"16x9"|"2x3"|"3x2"|"4x3"|"3x4"} - aspect ratio for <GridList.Tile.Primary>
+ */
+class GridList extends MaterialComponent {
+  get validationValuesByKey() {
+    return {
+      "with-icon-align": ["start", "end"],
+      "tile-gutter": [1],
+      "tile-aspect-ratio": ["1x1", "16x9", "2x3", "3x2", "4x3", "3x4"]
+    };
+  }
+
+  isValidValue(validationValues, testValue) {
+    return (
+      validationValues &&
+      validationValues.findIndex(val => val === testValue) >= 0
+    );
+  }
+
+  constructor() {
+    super();
+    this.componentName = "grid-list";
+    this._mdcProps = [
+      "header-caption",
+      "twoline-caption",
+      "tile-gutter",
```

Comment:
can we keep this `tile-gutter-1`? that ways it becomes super easy and less code.

Reply:
I guess. Since it is a number, I assumed this was something that could change in the future and at that point it would be a breaking change.